onos_ric_sdk_py/sdl.py

```diff
@@ -74,6 +75,96 @@ async def get_cell_ids(self, e2_node_id: str) -> List[str]:
         except GRPCError as e:
             raise ClientRuntimeError() from e
 
+    async def _get_cell_entity_id(self, e2_node_id: str, cell_id: str) -> int:
+        """
+        given e2_node_id and cell_id, returns entity id
+        raises ClientRuntimeError if cell_id is not found
+        """
+        if not self._ready:
+            raise ClientStoppedError()
+
+        client = TopoStub(self._topo_channel)
+        filters = Filters(
+            relation_filter=RelationFilter(
+                src_id=e2_node_id,
+                relation_kind=RanRelationKinds.CONTAINS.name.lower(),
+                target_kind="",
+            )
+        )
+
+        try:
+            response = await client.list(filters=filters)
+            for obj in response.objects:
+                if obj.entity.kind_id != RanEntityKinds.E2CELL.name.lower():
+                    continue
+
+                aspects = obj.aspects["onos.topo.E2Cell"].value
+                e2_cell = E2Cell().from_json(aspects)
+                if e2_cell.cell_object_id == cell_id:
+                    return obj.id
+        except GRPCError as e:
+            raise ClientRuntimeError() from e
+
+        raise ClientRuntimeError(
+            f"cannot find cell_id:{cell_id} in e2_node:{e2_node_id}"
+        )
+
+    async def get_cell_data(
+        self, e2_node_id: str, cell_id: str, key: str
+    ) -> Optional[bytes]:
+        """
+        get data referenced by key attached to a cell_id
+        """
+
+        if not self._ready:
+            raise ClientStoppedError()
+
+        entity_id = await self._get_cell_entity_id(e2_node_id, cell_id)
+
+        client = TopoStub(self._topo_channel)
+        try:
+            resp = await client.get(id=entity_id)
+        except GRPCError as e:
+            raise ClientRuntimeError() from e
+
+        type_data = resp.object.aspects.get(key)
+        if type_data is None:
+            return None
+
+        return type_data.value
+
+    async def set_cell_data(
+        self, e2_node_id: str, cell_id: str, key: str, data: bytes
```

Comment:
`data` needs to be `Optional[bytes]` if it can be `None`.

Reply:
I made the change to return `None` if cell id does not exist